### PR TITLE
first commit

### PR DESCRIPTION
**Pull request by RuslanIbIsh**

### CODE REVIEW

**Review by Ilya-brinchuk — APPROVED** [review]

File: src/main/java/com/iri/multithreading/service/ThreadTwo.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package com.iri.multithreading.service;
+
+import org.apache.log4j.Logger;
+
+public class ThreadTwo implements Runnable {
```
Comment (RomanDubovskyi) [inline src/main/java/com/iri/multithreading/service/ThreadTwo.java:5]:
```suggestion
public class ImplementingRunnable implements Runnable {
```

File: src/main/java/com/iri/multithreading/service/ThreadOne.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package com.iri.multithreading.service;
+
+import org.apache.log4j.Logger;
+
+public class ThreadOne extends Thread {
```
Comment (RomanDubovskyi) [inline src/main/java/com/iri/multithreading/service/ThreadOne.java:5]:
```suggestion
public class ExtendingThread extends Thread {
```

File: src/main/java/com/iri/multithreading/Main.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import com.iri.multithreading.service.ThreadOne;
+import com.iri.multithreading.service.ThreadTwo;
+
+public class Main {
+    public static void main(String[] args) {
+        Counter counter = new Counter();
+        ThreadOne threadOne = new ThreadOne(counter);
+        ThreadTwo threadTwo = new ThreadTwo(counter);
+        threadOne.start();
+        new Thread(threadTwo).start();
```
Comment (RomanDubovskyi) [inline src/main/java/com/iri/multithreading/Main.java:13]:
```suggestion
        Thread implmentingRunnable = new Thread(new ImplmmentingRunnable(counter));
        extendingThread.start();
        implmentingRunnable.start();
```

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/com/iri/multithreading/service/ThreadTwo.java (hunk 1 above)
Comment (RuslanIbIsh) [reply src/main/java/com/iri/multithreading/service/ThreadTwo.java:5]: done

File: src/main/java/com/iri/multithreading/service/ThreadOne.java (hunk 2 above)
Comment (RuslanIbIsh) [reply src/main/java/com/iri/multithreading/service/ThreadOne.java:5]: done

File: src/main/java/com/iri/multithreading/Main.java (hunk 3 above)
Comment (RuslanIbIsh) [reply src/main/java/com/iri/multithreading/Main.java:13]: done

File: src/main/java/com/iri/multithreading/Main.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package com.iri.multithreading;
+
+import com.iri.multithreading.service.Counter;
+import com.iri.multithreading.service.ExtendingThread;
+import com.iri.multithreading.service.ImplementingRunnable;
+
+public class Main {
+    public static void main(String[] args) {
+        Counter counter = new Counter();
+        ExtendingThread extendingThread = new ExtendingThread(counter);
```
Comment (RomanDubovskyi) [inline src/main/java/com/iri/multithreading/Main.java:10]:
```suggestion
        Thread extendingThread = new ExtendingThread(counter);
```
remember to use abstract types if it's possible

**Review by RomanDubovskyi — APPROVED** [review]

File: src/main/java/com/iri/multithreading/Main.java (hunk 4 above)
Comment (RuslanIbIsh) [reply src/main/java/com/iri/multithreading/Main.java:10]: done

**Pull request merged**